### ISSUE TITLE
How do you do transfer learning based on the models from the model zoo?

### ISSUE THREAD

**Issue by devedse** [AlexeyAB/darknet#7254]
I found the model zoo useful because I want to start training a model on my 3080. I decided that the `YOLOv4-Mish-416` would probably fit fine.

For the normal `yolov4.weights` there's also a `yolov4.conv.137` available. I can't find a similar file to the `YOLOv4-Mish-416` file though. Is there a way to generate this file yourself or obtain it somewhere else?

Next to that I also saw Scaled-YoloV4 also being mentioned a lot of times. Are the models from the model zoo actually using YoloV4 scaled or not?

**Comment by AlexeyAB** [comment]
Look at these examples how to generate your own pre-traineed weights like `yolov4.conv.137` from `yolov4.weights` https://github.com/AlexeyAB/darknet/blob/master/build/darknet/x64/partial.cmd

* Use https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v3_optimal/yolov4.conv.137 to train YOLOv4, YOLOv4-Mish, YOLOv4-SAM-Mish

* Use https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4-csp.conv.142 to train **Scaled-YOLOv4** `yolov4-csp.cfg`
* Use https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4x-mish.conv.166 to train **Scaled-YOLOv4** `yolov4x-mish.cfg`
* Use https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/cspx-p7-mish_hp.344.conv to train **Scaled-YOLOv4** https://github.com/AlexeyAB/darknet/blob/master/cfg/cspx-p7-mish.cfg

**Comment by devedse** [comment]
So how do you know what that last number is? E.g. 137 for yolov4, but what is it for leaky or Mish?

**Comment by AlexeyAB** [comment]
Use the number of layer prior to the first [yolo] layer.

**Comment by devedse** [comment]
Hmm, I tried this on the `yolov4.cfg`. I'm not experienced with darknet as of yet so I assume that everything in between brackets (`[...]`) is a layer. I counted all instances of that and came to a total of 140.

I assume the first one `[net]` shouldn't be counted but that still results in a count of 139. Which is 2 more then expected.

![image](https://user-images.githubusercontent.com/2350015/104818781-09ba3100-582a-11eb-8e7e-913a85fda2bc.png)

**Comment by devedse** [comment]
From what I read regarding scaled YOLO it seems that the CSP model might actually be one of the better ones.

However I don't think the scaled models are described in the model zoo?:
https://github.com/AlexeyAB/darknet/wiki/YOLOv4-model-zoo

Or am I misunderstanding some things?

**Comment by AlexeyAB** [comment]
Yes, YOLOv4-CSP it the best (but not for small objects): https://github.com/AlexeyAB/darknet/issues/7087#issuecomment-750924784

CSP model is described there: https://github.com/AlexeyAB/darknet/issues/7087

And there: https://alexeyab84.medium.com/scaled-yolo-v4-is-the-best-neural-network-for-object-detection-on-ms-coco-dataset-39dfa22fa982?source=friends_link&sk=c8553bfed861b1a7932f739d26f487c8

![1_epe7z6IqRuzOvPmTc00tpQ (1)](https://user-images.githubusercontent.com/4096485/104820799-662c4900-5848-11eb-9154-d6a945a6e5da.png)


----

To get layer number just run detection on any image:
![image](https://user-images.githubusercontent.com/4096485/104819423-78a18500-583e-11eb-9b01-26d2c32b2c31.png)

**Comment by devedse** [comment]
![chart_yolov4-csp-obj](https://user-images.githubusercontent.com/2350015/104860922-db257e80-592d-11eb-9f2e-713d011001d6.png)

@AlexeyAB , i have been training for about 24 hours and this is the result. Is this to be expected?
